### PR TITLE
Tracking for Plugin

### PR DESCRIPTION
**Pull request by rchougule**

### CODE REVIEW

File: src/main/java/com/browserstack/automate/ci/common/Tools.java [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
 import java.util.Arrays;
+import java.util.Locale;
 import java.util.regex.Pattern;
 
 public class Tools {
 
     public static final Pattern buildUrlPattern = Pattern.compile("(https?:\\/\\/[\\w-.]+\\/builds\\/\\w+)\\/sessions\\/\\w+");
+    public static final SimpleDateFormat readableDateFormat = new SimpleDateFormat("dd MMM yyyy, HH:mm");
```
Comment (Archish27) [inline src/main/java/com/browserstack/automate/ci/common/Tools.java]: READABLE_DATE_FORMAT

File: src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java [hunk 2]
```diff
@@ -164,7 +179,16 @@ private JSONObject convertSessionToJsonObject(Session session) {
...
 
-        sessionJSON.put(Constants.SessionInfo.CREATED_AT, session.getCreatedAt());
+        try {
+            Date sessionCreatedAt = Tools.SESSION_DATE_FORMAT.parse(session.getCreatedAt());
+            sessionJSON.put(Constants.SessionInfo.CREATED_AT, sessionCreatedAt);
+
+            String createdAtReadable = String.format("%s %s",
+                    Tools.readableDateFormat.format(sessionCreatedAt), "UTC");
+            sessionJSON.put(Constants.SessionInfo.CREATED_AT_READABLE, createdAtReadable);
+        } catch (ParseException ignored) {
```
Comment (Archish27) [inline src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java]: Are we not logging exception somewhere? Just swallow?

File: src/main/java/com/browserstack/automate/ci/jenkins/pipeline/BrowserStackReportStepExecution.java [hunk 3]
```diff
@@ -27,19 +28,26 @@ protected Void run() throws Exception {
         Run<?, ?> run = getContext().get(Run.class);
         TaskListener taskListener = getContext().get(TaskListener.class);
         PrintStream logger = taskListener.getLogger();
+        PluginsTracker tracker = new PluginsTracker();
```
Comment (Archish27) [inline src/main/java/com/browserstack/automate/ci/jenkins/pipeline/BrowserStackReportStepExecution.java]:
```suggestion
        final PluginsTracker tracker = new PluginsTracker();
```

File: src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java (hunk 2 above)
Comment (rchougule) [reply src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java]: its not really required actually. Such case shouldn't happen if we have session's data already. Will add the logging...

File: src/main/java/com/browserstack/automate/ci/jenkins/pipeline/BrowserStackReportStepExecution.java (hunk 3 above)
Comment (rchougule) [reply src/main/java/com/browserstack/automate/ci/jenkins/pipeline/BrowserStackReportStepExecution.java]: done

File: src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java (hunk 2 above)
Comment (rchougule) [reply src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java]: added the logging

File: src/main/java/com/browserstack/automate/ci/common/Tools.java (hunk 1 above)
Comment (rchougule) [reply src/main/java/com/browserstack/automate/ci/common/Tools.java]: done

**Review by Archish27 — APPROVED** [review]
Nice 🚀

**Pull request merged**